docs/bsp.md

```diff
@@ -1207,6 +1208,90 @@ Clients can use this information to reconstruct the tree of sbt meta builds. The
 `parent` information can be defined from `children` but it's provided by the server to
 simplify the data processing on the client side.
 
+## Connecting to a BSP Server
+
+The Build Server Protocol defines a standard to connect to BSP servers.
+This standard has been designed such that:
+
+1. Clients don't need to have specific support for a server to be able to
+   connect to it.
+1. Clients can connect to build tools installed both at the machine or at the
+   project level.
+1. Different versions of the same build tool can have different connection
+   mechanisms.
+
+The following protocol to discover and connect to BSP servers is inspired by
+[Jupyter's
+Kernelspec](https://jupyter-client.readthedocs.io/en/stable/kernels.html#kernelspecs)
+and incorporates changes required to deal with  requirements above.
+
+### The BSP Connection Details
+
+The BSP connection details are modelled in a JSON object generated by a build
+tool command. The object contains the following fields to allow clients to
+connect to the server.
+
+```scala
+trait BspConnectionDetails {
+  /** The name of the build tool. */
+  def name: String
+  /** The version of the build tool. */
+  def version: String
+  /** A collection of languages supported by this BSP server. */
+  def languages: List[String]
+  /** A sequence of commands to connect to the BSP server that
+    * can be executed via system processes. */
+  def argv: List[String]
+}
+```
+
+Every build tool supporting BSP must implement a build-tool-specific command to
+generate the BSP connection details.
+
+When such command is invoked, the build tool generates the connection details
+and writes them to a file located in a standard BSP directory
+`<workspace-directory>/.bsp`. The `.bsp/` directory must be at the root of the
+workspace base directory and the names of the files must be unique per build
+tool and version.
```

Comment:
We can remove this requirement, it's not a big deal.

Reply:
The wording jsut specifies that the name has to be unique per tool/version. How the tool achieves that is up to the tool. If there's one global installation it could just overwrite the old json, if there's parallel installations (eg development versions) it can create them with any arbitrary filename.